consensus/hotstuff/votecollector/combined_vote_processor_v2.go

```diff
@@ -51,8 +52,26 @@ func (f *combinedVoteProcessorFactoryBaseV2) Create(block *model.Block) (hotstuf
 		return nil, fmt.Errorf("could not create aggregator for staking signatures: %w", err)
 	}
 
-	rbRector := &signature.RandomBeaconReconstructor{} // TODO: initialize properly when ready
+	publicKeyShares := make([]crypto.PublicKey, 0, len(allParticipants))
+	dkg, err := f.committee.DKG(block.BlockID)
+	if err != nil {
+		return nil, fmt.Errorf("could not get DKG info at block %v: %w", block.BlockID, err)
+	}
+	for _, participant := range allParticipants {
+		pk, err := dkg.KeyShare(participant.NodeID)
+		if err != nil {
+			return nil, fmt.Errorf("could not get random beacon key share for %x: %w", participant.NodeID, err)
+		}
+		publicKeyShares = append(publicKeyShares, pk)
+	}
+
+	threshold := msig.RandomBeaconThreshold(int(dkg.Size()))
+	randomBeaconInspector, err := signature.NewRandomBeaconInspector(dkg.GroupKey(), publicKeyShares, threshold, msg)
+	if err != nil {
+		return nil, fmt.Errorf("could not create random beacon inspector: %w", err)
+	}
```

Comment:
This is the code that I think it could go inside `NewRandomBeaconReconstructor`.